man/subgid.5.xml

```diff
@@ -86,6 +89,42 @@
 
   </refsect1>
 
+  <refsect1 id='options'>
+    <title>OPTIONS</title>
+
+    <para>
+      Options are comma-separated. Empty options are ignored, and if the
+      field is missing entirely it is treated as an empty string. Attempting
+      to use an unknown option will cause <command>newgidmap</command> to emit
+      an error. Setting options from the same option-set multiple times in a
+      single entry will result in the last option specified taking precedence.
+    </para>
+
+    <variablelist>
+      <varlistentry>
+        <term><option>allow_setgroups</option> (default),
```

Comment:
Hm, is this the right place to put this kind of thing though?  Maybe it is, but:

The only case I know of for deny-setgroups is when an administrator has placed a user in a group with a negative acl, to prevent that user from accessing some resource.

In that case, does it make more sense to place a sort of "group_locked" option on the group itself in /etc/group?  Then when newgidmap runs, it checks whether the target pid is in any locked group, and if so sets deny_setgroup.

Placing the deny/allow_setgroup variable in the gid mapping seems a step removed, and might cause an admin to not notice that "hey, the user being allowd to grab those gids should have been locked into group nos3critdocs.

Reply:
Hmm, I hadn't though of this point.

You're quite right that most users of negative ACLs are done with groups (and I think it does make intuitive sense to be able to say "if you are in a bad group then you cannot use `setgroups`"), but I have a couple of questions about edge cases:

* Should an administrator be able to override this decision for particular users or groups? Should that also be in `/etc/groups` (or `/etc/passwd`)?

* What if it's a numeric group (so not one registered in `/etc/groups`)? Is the solution to just tell people "tough luck, you should register your group"?

* This is a bit of an edge condition (and is an edge condition that actually isn't really easy to handle with the current implementation either), but how will this affect containers on the system that are also using group blacklisting? While `/etc/sub*` are global, if you have containers using allocated portions of the UID/GID space then you would expect that a user shouldn't be able to use `newgidmap` to drop the negative ACLs for a different group set than the one you are joining? Now, this is quite a weird case because it would require you to have a quite strange ACL setup -- so maybe this is a use case we don't care about?

Reply:
> Should an administrator be able to override this decision for particular users or groups? Should that also be in /etc/groups (or /etc/passwd)?

I don't really think so, as you either want to deny the group certain access or you don't, but my gut says there are more subtle cases.  So let's say yes.

> What if it's a numeric group (so not one registered in /etc/groups)? Is the solution to just tell people "tough luck, you should register your group"?

I think requiring a name makes sense.  Do you?

> how will this affect containers on the system that are also using group blacklisting? 

Hmm - that applies to your approach too right?

It really seems like this requires a new kernel functionality to make the most sense - a new per-process 'lockedgrp' from which groups can never be removed.  Otherwise all we can really do in the parent container is say "never let the child container do a setgroup" (right?  or have i confused myself on this topic once again?)